### ISSUE TITLE
false warning by Nullness Checker

### ISSUE THREAD

**Issue by nikshinde1996** [typetools/checker-framework#1192]
Consider following code,
```
public class FileTest {
	public  File[] getPartitions() {
		File[] partitions = File.listRoots();
		return partitions;
	}
}
```

The Nullness Checker shows Following warning.
```
FileTest.java:7: error: [return.type.incompatible] incompatible types in return.
                return partitions;
                       ^
  found   : @Initialized @NonNull File @Initialized @Nullable []
  required: @Initialized @NonNull File @Initialized @NonNull []
1 error
```
why is this error reported by checker ? How should I deal with it ?

Thanks,

**Comment by mernst** [comment]
This is not a false positive warning.  The Nullness Checker is behaving exactly as intended.

The error is reported because `getPartitions` is declared to return a non-null value, but `listRoots` may return a null value.  You need to either change the declaration of the return value of `getPartitions`, or else add logic inside `getPartitions` so that it returns a non-null value that is consistent with its declaration.

(When you report a bug, it is good to give your reasoning for why it is a bug.)
